portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -93,19 +98,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = request.getParameter("username");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+
     String comment = request.getParameter("comment");
     String time = DateTimeFormatter.ISO_DATE_TIME.format(ZonedDateTime.now(ZoneId.of("UTC")));
 
     Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("name", name);
+
+    commentEntity.setProperty("name", userService.getCurrentUser().getUserId());
     commentEntity.setProperty("comment", comment);
     commentEntity.setProperty("time", time);
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
-    String url = "/comments.html?numComments=" + numDisplay + "&commentSort=" + sortOrder;
+    String url = "/comments.html";
+    try {
+      url = appendQuery(url, "commentSort=" + sortOrder).toString();
+      url = appendQuery(url, "numComments=" + numDisplay).toString();
+    } catch (Exception e) {}
```

Comment:
It seems risky to me to have a general Exception catch with no logic.  What kind of Exception do you expect to occur here?  Can we catch only that specific Exception?

Reply:
Yes, there is a URISyntaxException I can catch instead.